### PR TITLE
Add type checks to the compiler and fix typing issues

### PR DESCRIPTION
**Pull request by jonkoops**

### CODE REVIEW

File: webpack.common.ts [hunk 1]
```diff
@@ -271,6 +272,14 @@ export function createConfig(additionalOptions: CreateConfigOptions): Configurat
             : false,
       }),
       ...(!options.singleBuild ? [new HtmlWebpackMultiBuildPlugin()] : []),
+      new ForkTsCheckerWebpackPlugin({
```
Comment (royderks) [inline webpack.common.ts:275]: Do we also need this in production? Or is it sufficient to include it in develop only?

Comment (jonkoops) [reply webpack.common.ts:275]: I think it makes sense to include this in production as well as we want to run these builds on the CI.

**Comment by jonkoops** [comment]
Atlas has a lot of type errors, so I'll update this PR with some fixes, `ts-ignore` and `any` types to make the build pass. Will do this when I have some time as this is not our main concern right now.

**Comment by jonkoops** [comment]
I'll be opening seperate PRs to update the type compatibility and rebasing this PR with develop until the build succeeds.

File: src/app/pages/MapPage/MapPage.tsx [hunk 2]
```diff
@@ -137,15 +130,7 @@ const MapPage: React.FC = () => {
             />
           ))}
           {nonTmsLayers.map(({ url, overlayOptions: options, id }) => (
-            <NonTiledLayer
-              key={id}
-              url={url}
-              options={options}
-              events={{
```
Comment (chrisvanmook) [inline src/app/pages/MapPage/MapPage.tsx:144]: why is this removed?

File: src/app/pages/MapPage/MapPage.tsx [hunk 3]
```diff
@@ -172,15 +157,17 @@ const MapPage: React.FC = () => {
                 )}
                 {!showDrawTool && !detailUrl && location && (
                   <PointSearchPanel
-                    {...{
```
Comment (chrisvanmook) [inline src/app/pages/MapPage/MapPage.tsx:175]: why is this replaced? something to do with typechecking?

**Comment by chrisvanmook** [comment]
I presume that you tested the app thoroughly right? I can't really estimate what the impact is of all these changes...

File: src/app/components/CardList/CardList.tsx [hunk 4]
```diff
@@ -54,18 +53,15 @@ const CardList: React.FC<CardListProps> = ({ title, loading, results }) => (
             coverImage,
           }) => (
             <EditorialCard
-              {...{
-                ...linkProps, // !important: linkProps also contains `title`
```
Comment (royderks) [inline src/app/components/CardList/CardList.tsx:58]: Did you check if the title is still displayed correct?

Comment (jonkoops) [reply src/app/components/CardList/CardList.tsx:58]: Yes, the `title` prop is overwritten further down.

File: src/app/pages/MapPage/MapPage.tsx (hunk 3 above)
Comment (jonkoops) [reply src/app/pages/MapPage/MapPage.tsx:175]: Yeah, if you spread an object you cannot easily see which individual properties are invalid.

File: src/app/pages/MapPage/MapPage.tsx (hunk 2 above)
Comment (jonkoops) [reply src/app/pages/MapPage/MapPage.tsx:144]: I checked `NonTiledLayer` and it looks like it does not have an `events` prop (see: https://github.com/Amsterdam/amsterdam-react-maps/blob/master/packages/arm-nontiled/src/components/NonTiledLayer.tsx#L6-L10)

**Comment by jonkoops** [comment]
> I presume that you tested the app thoroughly right? I can't really estimate what the impact is of all these changes...

Yes, I've tested almost all of the code except small changes and it seems to work all right.
